include/otp_keystore.h

```diff
@@ -0,0 +1,72 @@
+/* otp_keystore.h
+ *
+ * Helper for storing/retrieving Trust Anchor to/from OTP flash
+ *
+ *
+ * Copyright (C) 2024 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#ifndef OTP_KEYSTORE_H
+#define OTP_KEYSTORE_H
+
+#if defined(FLASH_OTP_KEYSTORE) && !defined(WOLFBOOT_NO_SIGN)
+/* Specific includes for supported targets
+ * (needed for OTP_SIZE)
+ */
+#ifdef TARGET_stm32h7
+    #include "hal/stm32h7.h"
+#elif defined TARGET_stm32h5
+    #include "hal/stm32h5.h"
+#else
+    #error "Unsupported target for OTP"
+#endif
+
+#include "keystore.h"
+
+#define OTP_HDR_SIZE 16
+
+struct __attribute__((packed)) wolfBoot_otp_hdr {
```

Comment:
Eventually I'd like to see our use of PACKED come from a macro, so it's not just GCC.

```
#if (defined(__IAR_SYSTEMS_ICC__) && (__IAR_SYSTEMS_ICC__ > 8)) || \
        defined(__GNUC__)
    #define WOLFBOOT_PACK __attribute__ ((packed))
#else
    #define WOLFBOOT_PACK
#endif
```